prow/pod-utils/clone/clone.go

```diff
@@ -0,0 +1,86 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package clone
+
+import (
+	"bytes"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"k8s.io/test-infra/prow/kube"
+)
+
+func Run(refs kube.Refs, dir string) Record {
+	repositoryURL := fmt.Sprintf("https://github.com/%s/%s.git", refs.Org, refs.Repo)
+	cloneDir := fmt.Sprintf("%s/src/github.com/%s/%s", dir, refs.Org, refs.Repo)
+	record := Record{Refs: refs}
+
+	commands := []cloneCommand{
+		func() (string, string, error) {
+			return fmt.Sprintf("os.MkdirAll(%s, 0755)", cloneDir), "", os.MkdirAll(cloneDir, 0755)
+		},
+	}
+
+	commands = append(commands, shellCloneCommand(cloneDir, "git", "init"))
```

Comment:
This handily accommodates the case where the repo has already been cloned in the same location and we just update it as well as the case where we do a net-new clone.